implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicMeterRegistry.java

```diff
@@ -105,181 +102,10 @@ public void start(ThreadFactory threadFactory) {
-            // NOT when publishing an eventType per Meter/metric name
-            int size = attributes.length;
-            Attribute[] newAttrs = Arrays.copyOf(attributes, size + 2);
-
-            String name = id.getConventionName(config().namingConvention());
-            newAttrs[size] = new Attribute("metricName", name);
-            newAttrs[size + 1] = new Attribute("metricType", id.getType().toString());
-
-            return event(id, Tags.empty(), newAttrs);
-        }
-        return event(id, Tags.empty(), attributes);
-    }
-
-    private String event(Meter.Id id, Iterable<Tag> extraTags, Attribute... attributes) {
-        StringBuilder tagsJson = new StringBuilder();
-
-        for (Tag tag : getConventionTags(id)) {
-            tagsJson.append(",\"").append(escapeJson(tag.getKey())).append("\":\"").append(escapeJson(tag.getValue())).append("\"");
-        }
-
-        NamingConvention convention = config().namingConvention();
-        for (Tag tag : extraTags) {
-            tagsJson.append(",\"").append(escapeJson(convention.tagKey(tag.getKey())))
-                    .append("\":\"").append(escapeJson(convention.tagValue(tag.getValue()))).append("\"");
-        }
-
-        String eventType = getEventType(id);
-        
-        return Arrays.stream(attributes)
-                .map(attr ->
-                        (attr.getValue() instanceof Number)
-                            ? ",\"" + attr.getName() + "\":" + DoubleFormat.wholeOrDecimal(((Number)attr.getValue()).doubleValue())
-                            : ",\"" + attr.getName() + "\":\"" + convention.tagValue(attr.getValue().toString()) + "\""
-                )
-                .collect(Collectors.joining("", "{\"eventType\":\"" + escapeJson(eventType) + "\"", tagsJson + "}"));
-    }
-
-    private String getEventType(Meter.Id id) {
-        if (config.meterNameEventTypeEnabled()) {
-            return id.getConventionName(config().namingConvention());
-        } else {
-            return config.eventType();
-        }
-    }
-    
-    private void sendEvents(String insightsEndpoint, Stream<String> events) {
-        try {
-            AtomicInteger totalEvents = new AtomicInteger();
-
-            httpClient.post(insightsEndpoint)
-                    .withHeader("X-Insert-Key", config.apiKey())
-                    .withJsonContent(events.peek(ev -> totalEvents.incrementAndGet()).collect(Collectors.joining(",", "[", "]")))
-                    .send()
-                    .onSuccess(response -> logger.debug("successfully sent {} metrics to New Relic.", totalEvents))
-                    .onError(response -> logger.error("failed to send metrics to new relic: http {} {}", response.code(), response.body()));
-        } catch (Throwable e) {
-            logger.warn("failed to send metrics to new relic", e);
-        }
+        clientProvider.publish(this, getMeters());
```

Comment:
Why pass both the MeterRegistry and the list of meters? We can get the latter from the first, and the Http implementation doesn't use the passed list of meters.

Reply:
got it